activity/activity_pool.go

```diff
@@ -133,6 +138,8 @@ func (pool *activityPoolImpl) initWorkers(ctx context.Context, temporal client.T
 	pool.tWorkers = make([]worker.Worker, 0)
 
 	for i := 0; i < len(workerInfo); i++ {
+		// set the graceful timeout for the worker
+		workerInfo[i].Options.WorkerStopTimeout = pool.graceTimeout
```

Comment:
Unfortunately no, because thus we end with a nonconsistent state. For example, if a user sets a timeout to 10s, but the general RR timeout would be 5s, RR kills the process without regard to the user value.